src/components/viewer/report-editor/Report.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import styles from './ReportEditor.scss';
+import { AnnotationAPIModel, Reasons } from 'api/annotations';
+import { PPScopeClass } from '../../../class_consts';
+
+interface IReportProps {
+  annotation: AnnotationAPIModel;
+  onCancel: (e) => void;
+  onSubmit: (reason: Reasons, comment: string) => void;
+}
+
+interface IReportState {
+  reason: Reasons;
+  comment: string;
+  showReasonError: boolean;
+}
+
+export default class Report extends React.Component<Partial<IReportProps>, Partial<IReportState>> {
+  constructor(props: IReportProps) {
+    super(props);
+    this.state = {};
+  }
+
+  handleInputChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const target = e.currentTarget;
+    const stateUpdate = {
+      [target.name]: target.value,
+      showReasonError: target.name !== 'reason' && this.state.showReasonError,
+    };
+    this.setState(stateUpdate);
+  }
+
+  getRadioId(reason: Reasons) {
```

Comment:
Do we need `id` for anything?

Reply:
To labels pair with inputs, so that you can check the input by clicking on the label.